centreon/www/include/configuration/configObject/timeperiod/DB-Func.php

```diff
@@ -568,3 +386,262 @@ function createTimePeriodsExceptions(array $params): void
     $statement->bindValue(':timeperiod_id', (int) $params['timeperiod_id'], \PDO::PARAM_INT);
     $statement->execute();
 }
+
+// ----------------- API CALLS --------------------
+
+/**
+ * Create a new timeperiod form formData.
+ *
+ * @param array<mixed> $ret
+ *
+ * @return int|null
+ */
+function insertTimePeriodInAPI(array $ret = []): int|null
+{
+    global $form, $basePath;
+
+    $formData = $ret === [] ? $form->getSubmitValues() : $ret;
+
+    try {
+        return insertTimeperiodByApi($formData, $basePath);
+    } catch (Throwable $th) {
+        CentreonLog::create()->error(
+            logTypeId: CentreonLog::TYPE_BUSINESS_LOG,
+            message: "Error while inserting timeperiod by api : {$th->getMessage()}",
+            customContext: ['form_data' => $formData, 'base_path' => $basePath],
+            exception: $th
+        );
+
+        echo "<div class='msg' align='center'>" . _($th->getMessage()) . '</div>';
```

Comment:
Are you sure you want to send the exception  message back to the user ? It is not often user-friendly :)

Reply:
seen together it's ok